src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -3300,7 +3300,78 @@ LogicalResult ONNXRandomUniformLikeOp::inferShapes(
 
 LogicalResult ONNXRangeOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
-  return emitError(NOT_IMPLEMENTED_MESSAGE);
+  // Cannot infer shape if no shape tensor is specified.
```

Comment:
Can you add a TODO if some of the restrictions are simplifications to get a PR quickly... i.e. be explicit below which restrictions should be lifted to be standard compliant. Same with dynamic sizes, if supported. Tx